File: src/main/java/no/nav/arbeidsgiver/min_side/services/tokenExchange/TokenExchangeClientStub.java
package no.nav.arbeidsgiver.min_side.services.tokenExchange;

import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

@Profile({"labs"})
@Component
public class TokenExchangeClientStub implements TokenExchangeClient {
    @Override
    public TokenXToken exchange(String subjectToken, String audience) {
        var tokenXToken = new TokenXToken();
        tokenXToken.access_token = "fake-access-token";
        tokenXToken.token_type = "fake-token-type";
        tokenXToken.issued_token_type = "fake-issued-token-type";
        tokenXToken.expires_in = 3600;
        return tokenXToken;
    }
}

File: src/main/kotlin/no/nav/arbeidsgiver/min_side/services/tokenExchange/TokenExchangeClientStub.kt
package no.nav.arbeidsgiver.min_side.services.tokenExchange

import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

@Profile("labs")
@Component
class TokenExchangeClientStub : TokenExchangeClient {
    override fun exchange(subjectToken: String, audience: String): TokenXToken {
        return TokenXToken(
            access_token = "fake-access-token",
            token_type = "fake-token-type",
            issued_token_type = "fake-issued-token-type",
            expires_in = 3600,
        )
    }
}